tests/utils/test_decorators.py

```diff
@@ -0,0 +1,41 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from airflow.decorators import task
+
+
+def test_remove_skip_if_decorator():
+    @task.skip_if(lambda context: True)
+    @task.virtualenv()
+    def f(): ...
+
+    xcom_arg = f()
+    source = xcom_arg.operator.get_python_source()
+
+    assert "skip_if" not in source
+
+
+def test_remove_run_if_decorator():
+    @task.run_if(lambda context: True)
+    @task.virtualenv()
+    def f(): ...
+
+    xcom_arg = f()
+    source = xcom_arg.operator.get_python_source()
+
+    assert "run_if" not in source
```

Comment:
Is it possible for this to test what `source` actually contains, instead of just testing it does _not_ contain `"run_if"`?

Also, since we are creating a test, can we also test other decorators too?

Reply:
I also wonder if it’s possible to create some future-proof checks so we don’t have this problem again if new decorators are added in the future.

Reply:
I think it might work with `ast`, but I'm not sure.
I'll try it and get back to you.

Reply:
I've fixed the test, but I'm not satisfied. 
The function that clears the decorator relies on `custom_operator_name`, which makes it tricky to apply the decorator dynamically.

Reply:
I agree, this feels too flimsy. Let’s remove the test file and settle.

Reply:
I think the current suite of tests is fine.